### DISCUSSION TITLE
Write to delta format?

### DISCUSSION THREAD

**Discussion by GabrielKuka** [duckdb/duckdb#8949] [General]
I have made some data transformations in duckdb and I have a resulting VIEW. I was wondering whether it is possible to write this VIEW delta format.

**Comment by djouallah** [comment]
use delta rs, convert your view to an arrow table then go from there

https://delta-io.github.io/delta-rs/python/usage.html#writing-delta-tables

**Reply by GabrielKuka** [reply]
I try to convert my view to arrow table like so:
```py
arrow_table = con.view("MY_VIEW").arrow()
```
However, my RAM fills up and my OS freezes. My data is at 92GB approx. 
I have 32GB of RAM. I have set the memory limit to 16 for duckdb:
```py
con.execute("PRAGMA memory_limit=16gb")
```
Is there a way this can be done without filling up all of RAM?

NOTE: I'm running duckdb 0.8.1

**Reply by Tishj** [reply]
I think you want to produce a record batch reader instead, so the process can be done streaming
`write_deltalake` seems to have support for this: <https://delta-io.github.io/delta-rs/python/api_reference.html#deltalake.write_deltalake>

You would produce this from duckdb using `.fetch_record_batch()`

**Reply by GabrielKuka** [reply]
Interesting idea. How can I achieve this? 

I ran this code:
```py
batch_reader = con.execute("SELECT * FROM MY_VIEW").fetch_record_batch()
```
where con is:
```py
con = duckdb.connect(":memory:")
```

My RAM still fills up and everything freezes. I'm new to duckdb, so I'm not sure where I'm going wrong with this.

**Reply by Tishj** [reply]
Try `sql` instead of execute, `execute` has to materiaize everything ahead of time

**Reply by GabrielKuka** [reply]
I tried this option but I get this error:
```py
AttributeError: This relation does not contain a column by the name of 'fetch_record_batch'
```

**Reply by Mause** [reply]
The function on the relation is instead named `record_batch`

**Reply by GabrielKuka** [reply]
I tried ```record_batch()``` instead of ```fetch_record_batch()```, but RAM fills up again. Not sure why this happens because I've also set the memory limit to 16gb while having 32gb in total.

**Reply by djouallah** [reply]
unfortunately, converting to arrow works very well for small data but I am trying with 200 M rows it is extremely slow, exporting to parquet is substantially faster
